src/python/WMCore/WorkQueue/DataStructs/WorkQueueElement.py

```diff
@@ -213,31 +217,34 @@ def statusMetrics(self):
     def passesSiteRestriction(self, site):
         """Takes account of white & black list, and input data to work out
         if site can run the work"""
-        # Don't check anything if trustSitelists is enabled and SiteWhitelist is empty or site is 
-        # is site whitelist
-        if self['NoLocationUpdate'] and \
-           (not self['SiteWhitelist'] or site in self['SiteWhitelist']):
+        # Site list restrictions
+        if self['SiteWhitelist'] and site not in self['SiteWhitelist']:
+            return False
+        if site in self['SiteBlacklist']:
+            return False
+        # Trust both input and pileup location (TrustSitelists+TrustPUSiteliss flag)
+        # provided the site whitelist is empty or it's whitelisted
+        if hasattr(self, 'NoLocationUpdate') and self['NoLocationUpdate']:
             return True
-        # data restrictions - all data must be present at site
-        for locations in self['Inputs'].values():
-            if site not in locations:
-                return False
-        # Parent data as well
-        if self['ParentFlag']:
+
+        # input data restrictions (TrustSitelists flag)
+        if hasattr(self, 'NoInputUpdate') and self['NoInputUpdate'] is False:
```

Comment:
I think this has to be self[''NoInputUpdate'], 'NoInputUpdate' is not attr so it will always return False. Same for NoLocationUpdate as well but that will be removed eventually, right?